src/components/Username/Username.tsx

```diff
@@ -15,16 +17,27 @@ interface IUsernameProps {
  * @param length - The target length for usernames that are too long, by default 20
  */
 export const Username = ({ text, length = 20 }: IUsernameProps) => {
-  const removeKeyword = 'service-account-';
-  const resultText = text.replace(removeKeyword, '').substring(0, length);
+  const resultText = truncateUsername(text, length);
 
-  return text !== resultText ? (
+  return resultText.length !== text.length ? (
     <TooltipWrapper tooltip={text}>
       <span>
-        {text.startsWith(removeKeyword) && <Label isCompact>SA</Label>} {resultText}
+        {text.startsWith(serviceAccountKeyword) && <Label isCompact>SA</Label>} {resultText}
+        <>..</>
       </span>
     </TooltipWrapper>
   ) : (
     <>{text}</>
   );
 };
+
+const truncateUsername = (username: string, length: number): string => {
+  const truncatedUsername = username.replace(serviceAccountKeyword, '').substring(0, length);
+
+  // prettify the result
```

Comment:
[optional] `return truncatedUsername.replace(/-$/, '');` would be enough, or even more generalized regex `/-+$/`, so finally:

```ts
const truncateUsername = (username: string, length: number): string => username.replace(serviceAccountKeyword, '').substring(0, length).replace(/-+$/, '');
```